File: app/src/main/java/com/iita/akilimo/models/ComputeRequest.java

package com.iita.akilimo.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import lombok.Data;

@JsonPropertyOrder({
        "CMP",
        "FCY",
        "FR",
        "IC",
        "PP",
        "SPH",
        "SPH",
        "ureaBagWt",
        "ureaCostperBag",
        "ureaavailable",
        "MOPBagWt",
        "MOPCostperBag",
        "MOPavailable",
        "CANBagWt",
        "CANCostperBag",
        "CANavailable",
        "DAPBagWt",
        "DAPCostperBag",
        "DAPavailable",
        "NPK151515BagWt",
        "NPK151515CostperBag",
        "NPK151515available",
        "NPK171717BagWt",
        "NPK171717CostperBag",
        "NPK171717available",
        "NPK201010BagWt",
        "NPK201010CostperBag",
        "NPK201010available",
        "NafakaBagWt",
        "NafakaCostperBag",
        "Nafakaavailable",
        "SSPBagWt",
        "SSPCostperBag",
        "SSPavailable",
        "TSPBagWt",
        "TSPCostperBag",
        "TSPavailable",
        "PD",
        "PD_window",
        "HD",
        "HD_window",
        "area",
        "areaUnits",
        "cassPD",
        "cassUP",
        "cassUP_m1",
        "cassUP_m2",
        "cassUP_p1",
        "cassUP_p2",
        "cassUW",
        "cost_LMO_areaBasis",
        "cost_manual_harrowing",
        "cost_manual_ploughing",
        "cost_manual_ridging",
        "cost_tractor_harrowing",
        "cost_tractor_ploughing",
        "cost_tractor_ridging",
        "cost_weeding1",
        "cost_weeding2",
        "country",
        "email",
        "SMS",
        "fallowGreen",
        "fallowHeight",
        "fallowType",
        "harrowing",
        "intercrop",
        "lat",
        "lon",
        "maizePD",
        "maizeUP",
        "maizeUW",
        "sweetPotatoPD",
        "sweetPotatoUW",
        "sweetPotatoUP",
        "maxInv",
        "method_harrowing",
        "method_ploughing",
        "method_ridging",
        "nameSF",
        "ploughing",
        "problemWeeds",
        "ridging",
        "riskAtt",
        "saleSF",
        "tractor_harrow",
        "tractor_plough",
        "tractor_ridger",
        "userEmail",
        "userField",
        "userName",
        "userPhoneCC",
        "userPhoneNr",
        "fullPhoneNumber",
        "fertilizerList"
})
//@Builder(toBuilder = true)
//@Getter
@Data
public class ComputeRequest {

    @JsonProperty("country")
    private String country;

    @JsonProperty("lat")
    private double mapLat;

    @JsonProperty("lon")
    private double mapLong;

    @JsonProperty("area")
    private double fieldArea;

    @JsonProperty("areaUnits")
    private String areaUnits;

    @JsonProperty("intercropType")
    private String interCroppingType;

    @JsonProperty("intercrop")
    private boolean interCrop;

    @JsonProperty("IC")
    private boolean interCroppingRec;
    @JsonProperty("FR")
    private boolean fertilizerRec;
    @JsonProperty("PP")
    private boolean plantingPracticesRec;
    @JsonProperty("SPH")
    private boolean scheduledPlantingRec;
    @JsonProperty("SPH")
    private boolean scheduledHarvestRec;

    @JsonProperty("PD")
    private String plantingDate;
    @JsonProperty("PD_window")
    private int plantingDateWindow;

    @JsonProperty("HD")
    private String harvestDate;
    @JsonProperty("HD_window")
    private int harvestDateWindow;

    @JsonProperty("fallowType")
    private String fallowType;

    @JsonProperty("fallowHeight")
    private int fallowHeight;
    @JsonProperty("fallowGreen")
    private boolean fallowGreen;
    @JsonProperty("problemWeeds")
    private boolean problemWeeds;

    @JsonProperty("tractor_plough")
    private boolean tractorPlough;
    @JsonProperty("tractor_harrow")
    private boolean tractorHarrow;
    @JsonProperty("tractor_ridger")
    private boolean tractorRidger;

    @JsonProperty("cost_LMO_areaBasis")
    private String costLmoAreaBasis;
    @JsonProperty("cost_tractor_ploughing")
    private String costTractorPloughing;
    @JsonProperty("cost_tractor_harrowing")
    private String costTractorHarrowing;
    @JsonProperty("cost_tractor_ridging")
    private String costTractorRidging;

    @JsonProperty("cost_manual_ploughing")
    private String costManualPloughing;
    @JsonProperty("cost_manual_harrowing")
    private String costManualHarrowing;
    @JsonProperty("cost_manual_ridging")
    private String costManualRidging;
    @JsonProperty("cost_weeding1")
    private String costWeedingOne;
    @JsonProperty("cost_weeding2")
    private String costWeedingTwo;

    @JsonProperty("ploughing")
    private boolean ploughingDone;
    @JsonProperty("harrowing")
    private boolean harrowingDone;
    @JsonProperty("ridging")
    private boolean ridgingDone;

    @JsonProperty("method_ploughing")
    private String methodPloughing;
    @JsonProperty("method_harrowing")
    private String methodHarrowing;
    @JsonProperty("method_ridging")
    private String methodRidging;

    @JsonProperty("FCY")
    private int currentFieldYield;

    @JsonProperty("CMP")
    private String currentMaizePerformance;
    @JsonProperty("saleSF")
    private Boolean sellToStarchFactory;
    @JsonProperty("nameSF")
    private String starchFactoryName;
    @JsonProperty("cassPD")
    private String cassavaProduceType;

    @JsonProperty("cassUW")
    private int cassavaUnitWeight;
    @JsonProperty("cassUP")
    private double cassavaUnitPrice;

    @JsonProperty("cassUP_m1")
    private String cassUPM1;
    @JsonProperty("cassUP_m2")
    private String cassUPM2;
    @JsonProperty("cassUP_p1")
    private String cassUPP1;
    @JsonProperty("cassUP_p2")
    private String cassUPP2;

    @JsonProperty("maizePD")
    private String maizeProduceType;
    @JsonProperty("maizeUW")
    private String maizeUnitWeight;
    @JsonProperty("maizeUP")
    private String maizeUnitPrice;

    @JsonProperty("sweetPotatoPD")
    private String sweetPotatoProduceType;
    @JsonProperty("sweetPotatoUW")
    private String sweetPotatoUnitWeight;
    @JsonProperty("sweetPotatoUP")
    private String sweetPotatoUnitPrice;

    @JsonProperty("maxInv")
    private Double maxInvestment;

    @JsonProperty("SMS")
    private boolean sendSms;
    @JsonProperty("email")
    private boolean sendEmail;

    @JsonProperty("userPhoneCC")
    private String mobileCountryCode;
    @JsonProperty("userPhoneNr")
    private String mobileNumber;
    @JsonProperty("fullPhoneNumber")
    private String fullPhoneNumber;
    @JsonProperty("userName")
    private String userName;
    @JsonProperty("userEmail")
    private String userEmail;
    @JsonProperty("userField")
    private String fieldDescription;

    @JsonProperty("riskAtt")
    private int riskAttitude;
}

File: app/src/main/java/com/iita/akilimo/entities/ComputeRequest.kt
package com.iita.akilimo.entities

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Data

@Data
class ComputeRequest {

    @JsonProperty("country")
    var country: String? = null

    @JsonProperty("currency")
    var currency: String? = null

    @JsonProperty("lat")
    var mapLat: Double = 0.toDouble()

    @JsonProperty("lon")
    var mapLong: Double = 0.toDouble()

    @JsonProperty("area")
    var fieldArea: Double = 0.toDouble()

    @JsonProperty("areaUnits")
    var areaUnits: String? = null
    @JsonProperty("intercropType")
    var interCroppingType: String? = null

    @JsonProperty("intercrop")
    var interCrop: Boolean = false

    @JsonProperty("IC")
    var interCroppingRec: Boolean = false
    @JsonProperty("FR")
    var fertilizerRec: Boolean = false
    @JsonProperty("PP")
    var plantingPracticesRec: Boolean = false
    @JsonProperty("SPP")
    var scheduledPlantingRec: Boolean = false
    @JsonProperty("SPH")
    var scheduledHarvestRec: Boolean = false

    @JsonProperty("PD")
    var plantingDate: String? = null
    @JsonProperty("PD_window")
    var plantingDateWindow: Int = 0

    @JsonProperty("HD")
    var harvestDate: String? = null
    @JsonProperty("HD_window")
    var harvestDateWindow: Int = 0

    @JsonProperty("fallowType")
    var fallowType: String? = null

    @JsonProperty("fallowHeight")
    var fallowHeight: Int = 0
    @JsonProperty("fallowGreen")
    var fallowGreen: Boolean = false
    @JsonProperty("problemWeeds")
    var problemWeeds: Boolean = false

    @JsonProperty("tractor_plough")
    var tractorPlough: Boolean = false
    @JsonProperty("tractor_harrow")
    var tractorHarrow: Boolean = false
    @JsonProperty("tractor_ridger")
    var tractorRidger: Boolean = false

    @JsonProperty("cost_LMO_areaBasis")
    var costLmoAreaBasis: String? = null
    @JsonProperty("cost_tractor_ploughing")
    var costTractorPloughing: String? = null
    @JsonProperty("cost_tractor_harrowing")
    var costTractorHarrowing: String? = null
    @JsonProperty("cost_tractor_ridging")
    var costTractorRidging: String? = null

    @JsonProperty("cost_manual_ploughing")
    var costManualPloughing: String? = null
    @JsonProperty("cost_manual_harrowing")
    var costManualHarrowing: String? = null
    @JsonProperty("cost_manual_ridging")
    var costManualRidging: String? = null
    @JsonProperty("cost_weeding1")
    var costWeedingOne: String? = null
    @JsonProperty("cost_weeding2")
    var costWeedingTwo: String? = null

    @JsonProperty("ploughing")
    var ploughingDone: Boolean = false
    @JsonProperty("harrowing")
    var harrowingDone: Boolean = false
    @JsonProperty("ridging")
    var ridgingDone: Boolean = false

    @JsonProperty("method_ploughing")
    var methodPloughing: String? = null
    @JsonProperty("method_harrowing")
    var methodHarrowing: String? = null
    @JsonProperty("method_ridging")
    var methodRidging: String? = null

    @JsonProperty("FCY")
    var currentFieldYield: Int = 0

    @JsonProperty("CMP")
    var currentMaizePerformance: String? = null
    @JsonProperty("saleSF")
    var sellToStarchFactory: Boolean? = null
    @JsonProperty("nameSF")
    var starchFactoryName: String? = null
    @JsonProperty("cassPD")
    var cassavaProduceType: String? = null

    @JsonProperty("cassUW")
    var cassavaUnitWeight: Int = 0
    @JsonProperty("cassUP")
    var cassavaUnitPrice: Double = 0.toDouble()

    @JsonProperty("cassUP_m1")
    var cassUPM1: String? = null
    @JsonProperty("cassUP_m2")
    var cassUPM2: String? = null
    @JsonProperty("cassUP_p1")
    var cassUPP1: String? = null
    @JsonProperty("cassUP_p2")
    var cassUPP2: String? = null

    @JsonProperty("maizePD")
    var maizeProduceType: String? = null
    @JsonProperty("maizeUW")
    var maizeUnitWeight: String? = null
    @JsonProperty("maizeUP")
    var maizeUnitPrice: String? = null

    @JsonProperty("sweetPotatoPD")
    var sweetPotatoProduceType: String? = null
    @JsonProperty("sweetPotatoUW")
    var sweetPotatoUnitWeight: String? = null
    @JsonProperty("sweetPotatoUP")
    var sweetPotatoUnitPrice: String? = null

    @JsonProperty("maxInv")
    var maxInvestment: Double? = null

    @JsonProperty("SMS")
    var sendSms: Boolean = false
    @JsonProperty("email")
    var sendEmail: Boolean = false

    @JsonProperty("userPhoneCC")
    var mobileCountryCode: String? = null
    @JsonProperty("userPhoneNr")
    var mobileNumber: String? = null
    @JsonProperty("fullPhoneNumber")
    var fullPhoneNumber: String? = null
    @JsonProperty("userName")
    var userName: String? = null
    @JsonProperty("userEmail")
    var userEmail: String? = null
    @JsonProperty("userField")
    var fieldDescription: String? = null

    @JsonProperty("riskAtt")
    var riskAttitude: Int = 0
}